### PR TITLE
Open new window for each project

### PR DESCRIPTION
**Pull request by TheCodez**
@amakaseev could you take a look, I'm getting some weird graphical glitches in the new window(e.g. atlas preview completely zoomed in)

### CODE REVIEW

**Comment by amakaseev** [comment]
I have looked on MacOS at everything well, opens in a new window
only there are small remarks:
now it is impossible to reopen the open project :( to reset not the necessary settings

    if (!_currentProjectFileName.isEmpty()) {
        if (fileName != _currentProjectFileName) {

to:     if (!_currentProjectFileName.isEmpty() && (fileName != _currentProjectFileName)) {

**Comment by TheCodez** [comment]
I'll try your changes

**Comment by amakaseev** [comment]
> I'm getting some weird graphical glitches in the new window
perhaps a problem that the atlas is generated in a thread?

**Comment by TheCodez** [comment]
That could be the cause. When I have an open project and open another one, the sprite preview in the new window is white until the atlas has been generated.

**Comment by amakaseev** [comment]
you have lost this line:

settings.setValue("spritePackerFileName", fileName);

File: SpriteSheetPacker/MainWindow.cpp [hunk 1]
```diff
@@ -294,6 +303,9 @@ void MainWindow::refreshPreview() {
 }
 
 void MainWindow::openSpritePackerProject(const QString& fileName) {
+    QSettings settings;
+    settings.setValue("spritePackerFileName", fileName);
```
Comment (TheCodez) [inline SpriteSheetPacker/MainWindow.cpp]: it's still there, hopefully thats the right place

**Comment by amakaseev** [comment]
try at first show
and then only openProjectFile

File: SpriteSheetPacker/MainWindow.cpp (hunk 1 above)
Comment (amakaseev) [inline SpriteSheetPacker/MainWindow.cpp]: I think yes... sorry

**Comment by TheCodez** [comment]
Already tried that, but doesn't change anything.

**Comment by amakaseev** [comment]
QPointer<MainWindow> wnd(new MainWindow());
            wnd->show();
            QCoreApplication::processEvents(QEventLoop::AllEvents, 100);
            wnd->openSpritePackerProject(fileName);

**Comment by amakaseev** [comment]
the white window has to be gone

**Comment by TheCodez** [comment]
Yep that fixed the white window. But the problem with the weird zoomed in preview still persists.

image

**Comment by amakaseev** [comment]
it is strange :(
at me so it is impossible to make... can because I have several scalevariants...
you have one scalevariant?

**Comment by TheCodez** [comment]
I just open the programm, press new project button to open a new window, and the new window shows this strange effect.

**Comment by TheCodez** [comment]
Also the zoomslider shouldn't have any impact on the zoom of the background brush? It should always look the same, only the atlas should be zoomed.

**Comment by amakaseev** [comment]
yes you are right, it a bug

**Comment by TheCodez** [comment]
That could potentially fix the problem

**Comment by amakaseev** [comment]
I haven't managed to correct him yet
if in the preview there are no sprites I want to hide it and to show the picture - a prompt as in a texturepacker :)

**Comment by TheCodez** [comment]
Then you can probably merge this pr, because the problem will most likely be gone once you change that

**Comment by TheCodez** [comment]
Interesting applying your change, I also get the zoom effect on the first window

**Comment by amakaseev** [comment]
also be convinced please that when closing windows destructors are called

**Comment by amakaseev** [comment]
> Interesting applying your change, I also get the zoom effect on the first window

I will correct it until remove a line from the constructor
109: refreshAtlas ();

**Comment by amakaseev** [comment]
it is improbable
many thanks!

**Pull request merged**